src/hooks/useENSRegistrationCosts.ts

```diff
@@ -94,37 +50,62 @@ export default function useENSRegistrationCosts({
       nativeAssetPrice
     );
 
-    const weiEstimatedTotalCost = add(
-      formattedEstimatedNetworkFee.wei,
-      rentPrice.toString()
-    );
-    const totalRegistrationCost = formatTotalRegistrationCost(
-      weiEstimatedTotalCost,
-      nativeCurrency,
-      nativeAssetPrice
-    );
-
-    return {
-      estimatedTotalRegistrationCost: totalRegistrationCost,
-    };
-  }, [accountAddress, duration, name, nativeCurrency, rentPrice]);
+    return formattedEstimatedNetworkFee;
+  }, [accountAddress, duration, name, nativeCurrency, rentPriceInWei]);
 
-  const { data, status } = useQuery(
-    rentPrice && [
-      'getRegistrationValuesEstimations',
-      duration,
-      name,
-      rentPrice,
+  const { data: estimatedNetworkFee, status } = useQuery(
+    Boolean(rentPriceInWei) && [
+      'getEstimatedNetworkFee',
+      [accountAddress, name, nativeCurrency, rentPriceInWei],
     ],
-    getRegistrationValuesEstimations
+    getEstimatedNetworkFee,
+    { cacheTime: 0 }
   );
 
+  const data = useMemo(() => {
+    const rentPricePerYearInWei = rentPrice?.perYear?.wei?.toString();
+    if (estimatedNetworkFee && rentPricePerYearInWei) {
+      const nativeAssetPrice = ethereumUtils.getPriceOfNativeAssetForNetwork(
+        Network.mainnet
+      );
+      const rentPrice = multiply(rentPricePerYearInWei, duration);
+      const estimatedRentPrice = formatRentPrice(
+        rentPrice,
+        duration,
+        nativeCurrency,
+        nativeAssetPrice
+      );
+
+      const weiEstimatedTotalCost = add(
+        estimatedNetworkFee.wei,
+        estimatedRentPrice.wei.toString()
+      );
+      const displayEstimatedTotalCost = addDisplay(
```

Comment:
hm I think we shouldn't be doing this, `estimatedTotalRegistrationCost.display` is not working for you? is being calculated in line 87

Reply:
leaving as a note to mention we're adding `currency.js` when we merge to develop it's a 200 lines file. from @jxom otherwise i can just copy and paste it into a util